pkg/validator/cluster.go

```diff
@@ -807,6 +807,49 @@ func addServiceType(
 	return valErrors, patches
 }
 
+// validateNamingScheme function validates the namingScheme for all K8s objects.
+// If the naming scheme is unspecified, check to see if the default namingScheme
+// is provided through kubedirector's config CR, otherwise use a global constant
+// for naming scheme. In either of those cases add an entry to PATCH spec for mutating
+// cluster CR.
+// If the naming scheme is defined, validate the naming scheme for correctness
+// and that the cluster name, role name combination does not exceed the default object
+// length limit for K8s objects (63). The max length limit is set to 56 to account for
+// the additional '-' and/or digit characters that might be included in the name.
+func validateNamingScheme(
+	cr *kdv1.KubeDirectorCluster,
+	appCR *kdv1.KubeDirectorApp,
+	valErrors []string,
+	patches []clusterPatchSpec,
+) ([]string, []clusterPatchSpec) {
+
+	if cr.Spec.NamingScheme == nil {
+		namingScheme := shared.GetDefaultNamingScheme()
+		cr.Spec.NamingScheme = &namingScheme
+		patches = append(
+			patches,
+			clusterPatchSpec{
+				Op:   "add",
+				Path: "/spec/namingScheme",
+				Value: clusterPatchValue{
+					ValueStr: cr.Spec.NamingScheme,
+				},
+			},
+		)
+	} else {
+		if *cr.Spec.NamingScheme != kdv1.CrNameRole && *cr.Spec.NamingScheme != kdv1.UID {
```

Comment:
Our schema validation should make this impossible.

Reply:
Isnt this in the validation path itself?

Reply:
This is our extra webhook-based validation here. The schema validation (handled by the K8s API server) on the other hand should already take care of enforcing the allowed values for the property, since it's declared like this:
```
            namingScheme:
              type: string
              pattern: '^UID$|^CrNameRole$'
```